roles/dns/README.md

```diff
@@ -0,0 +1,42 @@
+dns
+===
+
+Configure a DNS server serving IPs of all the nodes of the cluster
+
+Requirements
+------------
+
+None
+
+Role Variables
+--------------
+
+| Name | Mandatory / Optional | Description |
+|------|----------------------|-------------|
+| `dns_zones` | Mandatory | DNS zones in which we must find the hosts |
+| `dns_forwarders` | If not set, the DNS will be a recursive non-forwarding DNS server | DNS forwarders to delegate the requests for hosts outside of `dns_zones` |
+| `g_all_hosts` | Mandatory | Exhaustive list of hosts |
```

Comment:
We try not to use global variables directly in roles.

Instead, we prefer to pass in the global variable as a role variable.

So, something like this (untested):
```yaml
    - hosts: dns_hosts
      roles:
      - role: dns
        dns_forwarders: [ '8.8.8.8', '8.8.4.4' ]
        dns_zones: [ novalocal, openstacklocal ]
        dns_all_hosts: "{{ g_all_hosts }}"
```

That way we can pass in whatever we want to the role (makes it more generic).